core/server/worker/src/main/java/alluxio/worker/netty/DataServerHandler.java

```diff
@@ -63,45 +63,54 @@ public DataServerHandler(final AlluxioWorkerService worker) {
-        mBlockHandler.handleBlockWriteRequest(ctx, (RPCBlockWriteRequest) msg);
-        break;
-      case RPC_FILE_READ_REQUEST:
-        assert msg instanceof RPCFileReadRequest;
-        mUnderFileSystemHandler.handleFileReadRequest(ctx, (RPCFileReadRequest) msg);
-        break;
-      case RPC_FILE_WRITE_REQUEST:
-        assert msg instanceof RPCFileWriteRequest;
-        mUnderFileSystemHandler.handleFileWriteRequest(ctx, (RPCFileWriteRequest) msg);
-        break;
-      case RPC_ERROR_RESPONSE:
-        assert msg instanceof RPCErrorResponse;
-        LOG.error("Received an error response from the client: " + msg.toString());
-        break;
-      case RPC_READ_REQUEST:
-      case RPC_WRITE_REQUEST:
-      case RPC_RESPONSE:
-        assert msg instanceof RPCProtoMessage;
-        ctx.fireChannelRead(msg);
-        break;
-      default:
-        RPCErrorResponse resp = new RPCErrorResponse(RPCResponse.Status.UNKNOWN_MESSAGE_ERROR);
-        ctx.writeAndFlush(resp);
-        // TODO(peis): Fix this. We should not throw an exception here.
-        throw new IllegalArgumentException(
-            "No handler implementation for rpc msg type: " + msg.getType());
+    LOG.debug("Enter: {}", msg);
+    try {
+      switch (msg.getType()) {
+        case RPC_BLOCK_READ_REQUEST:
+          assert msg instanceof RPCBlockReadRequest;
+          mBlockHandler.handleBlockReadRequest(ctx, (RPCBlockReadRequest) msg);
+          break;
+        case RPC_BLOCK_WRITE_REQUEST:
+          assert msg instanceof RPCBlockWriteRequest;
+          mBlockHandler.handleBlockWriteRequest(ctx, (RPCBlockWriteRequest) msg);
+          break;
+        case RPC_FILE_READ_REQUEST:
+          assert msg instanceof RPCFileReadRequest;
+          mUnderFileSystemHandler.handleFileReadRequest(ctx, (RPCFileReadRequest) msg);
+          break;
+        case RPC_FILE_WRITE_REQUEST:
+          assert msg instanceof RPCFileWriteRequest;
+          mUnderFileSystemHandler.handleFileWriteRequest(ctx, (RPCFileWriteRequest) msg);
+          break;
+        case RPC_ERROR_RESPONSE:
+          assert msg instanceof RPCErrorResponse;
+          LOG.error("Received an error response from the client: " + msg.toString());
+          break;
+        case RPC_READ_REQUEST:
+        case RPC_WRITE_REQUEST:
+        case RPC_RESPONSE:
+          assert msg instanceof RPCProtoMessage;
+          ctx.fireChannelRead(msg);
+          break;
+        default:
+          RPCErrorResponse resp = new RPCErrorResponse(RPCResponse.Status.UNKNOWN_MESSAGE_ERROR);
+          ctx.writeAndFlush(resp);
+          IllegalArgumentException e =
+              new IllegalArgumentException("No handler implementation for " + msg.getType());
+          LOG.warn("Exit (Error): {}", msg, e);
+          // TODO(peis): Fix this. We should not throw an exception here.
+          throw e;
+      }
+    } catch (IOException e) {
+      LOG.warn("Exit (Error): {}", msg, e);
```

Comment:
Our other servers don't log every exception at warn level, so we should be consistent here. We may miss some important exceptions by moving to `DEBUG`, but the exception isn't entirely lost because we propagate it to the client, so we will be alright as long as either `debug` is enabled or client logs are available. The common case should be us working "correctly", so we would only see stack traces for exceptions which don't actually indicate a problem. I'd prefer `DEBUG` to move us towards a world where stack traces in our logs need be taken seriously.